### ISSUE TITLE
WsFederation troubleshooting section in docs

### ISSUE THREAD

**Issue by sethsnel** [dotnet/AspNetCore.Docs#30442]
# Issue description

I'm in the process of upgrading a .net Framework asp.net MVC application to .net 6 for my client. The client is using WsFederation for user authentication and managed. I've tried to follow this documentation for implementation: https://learn.microsoft.com/en-us/aspnet/core/security/authentication/ws-federation?view=aspnetcore-7.0. 

However I seem to get stuck and I miss additional documentation on how to troubleshoot my issues. I've already tried to using chatgpt and extensive google querying to resolve my issues. My progress in a nut shell:

- I've configured `AddWsFederation`.
- When I run the app I get redirected to the IdP.
- After login the IdP redirects me back.
- A wsfederation cookie is being set (I can see in browser dev tools)
- `_httpContextAccessor.HttpContext.User.Identity.Name` is still empty on redirect.

The issue looks similar to this one: https://github.com/dotnet/AspNetCore.Docs/issues/7653.

Is there perhaps anyone that has experience with wsfederation in .net 6? And willing to lend a hand. If my issue get's resolved and turns out to be common, we can perhaps improve the documentation on this.

# Software versions

Check the .NET target framework(s) being used, and include the version number(s).

* [x] .NET 6.0 or later
* [x] .NET Framework

<details>
<summary><strong>dotnet --info output</strong> or <strong>About VS info</strong></summary>

```console
Microsoft Windows [Version 10.0.22621.2283]
(c) Microsoft Corporation. All rights reserved.

C:\Users\sbb>dotnet --info
.NET SDK:
 Version:   7.0.203
 Commit:    5b005c19f5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.203\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  5.0.416 [C:\Program Files\dotnet\sdk]
  6.0.203 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.22 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```
</details>

**Comment by Rick-Anderson** [comment]
@chlowell do you have any debugging tips? cc @Tratcher @blowdart

**Comment by sethsnel** [comment]
@Rick-Anderson , I had a colleague pair programming with me yesterday. It turns out I had made two mistakes.
- I had set my `Wreply` to the index action of my home controller. But it turns out that I had to leave it default to `/signin-wsfed`. This endpoint is mentioned in the docs. But it did not occur to me that this endpoint was listened to by the WsFederation package after configuration in `Program.cs` (c.q. `Startup.cs`).
- Secondly, I found the 'real' issue. The `issuer` provided on redirect did not match what WsFederation was expecting. So I manually configured the issuer to accept the correct value:
```
options.TokenValidationParameters = new TokenValidationParameters {
    ValidIssuer = "http://localhost:55206/Account/SSO",
};
```

The `issuer` is not mentioned at all in this documentation: 'https://learn.microsoft.com/en-us/aspnet/core/security/authentication/ws-federation?view=aspnetcore-7.0'. Perhaps info on this for custom IdP could be added.
